### PR TITLE
Backport Jetpack 14.0 release changes

### PR DESCRIPTION
**Pull request by ice9js**
## Proposed changes:

Changelog edits ahead of Jetpack 14.0 release.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Proofread the changelog. Ensure it's clear and concise.

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
Looking good! 🚢

**Review by jeherve — DISMISSED** [review]
This works for me. I'm approving, but I think you'll need to bump the versions before we can merge this.

File: projects/plugins/jetpack/CHANGELOG.md [hunk 1]
```diff
@@ -2,141 +2,96 @@
...
-
-### Bug fixes
-- Google Fonts: Fix module not loading fully in certain scenarios. [#39977]
-- Social: Fix the Instagram maximum video length. [#39930]
-- Social: Prevent share status styles from affecting other independent elements on the page. [#39991]
-- Subscribe Block: Fix styling for the number of subscribers in the post editor sidebar. [#39929]
+## 14.0 - 2024-11-05
+### Major enhancements
+- Newsletters: Add email previews and a floating subscribe button. [#37722] [#39782]
+- Social: Add support for Bluesky. [#39659] [#39561]
```
Comment (tbradsha) [inline projects/plugins/jetpack/CHANGELOG.md:8]: Are these really "major enhancements"? 🤔

**Review by tbradsha — CHANGES_REQUESTED** [review]
Looks like there's a version mismatch that needs to be fixed.

File: projects/plugins/jetpack/CHANGELOG.md (hunk 1 above)
Comment (jeherve) [reply projects/plugins/jetpack/CHANGELOG.md:8]: I think they're new features that would qualify as major enhancements, we want folks to know about it.

**Comment by ice9js** [comment]
I backported all the changed from the release branch now in here as well which should resolve the version mismatch too.

**Review by jeherve — DISMISSED** [review]

**Review by tbradsha — APPROVED** [review]

**Pull request merged**